src/components/Navbar/Navbar.tsx

```diff
@@ -1,23 +1,56 @@
+import React, { useEffect, useState } from 'react';
 import { Dropdown } from '@/components/Dropdown/Dropdown';
+import Image from 'next/image';
 import Link from 'next/link';
 import { Logo } from '@/components/Icons/Logo';
-import React from 'react';
 import styles from './Navbar.module.scss';
 
 export function Navbar() {
+    const [show, setShow] = useState(true);
+    const [scrollY, setScrollY] = useState(0);
+
+    useEffect(() => {
+        const handleScroll = () => {
+            window.scrollY > scrollY ? setShow(false) : setShow(true);
+            setScrollY(window.scrollY);
+        };
+
+        const handleClick = (event: any) => {
+            if (!event.target.matches('#navbar, #navbar *')) {
+                setShow(false);
+            }
+        };
+
+        const body = document.querySelector('body');
+        if (!body) {
+            return;
+        }
+        const options = { passive: true } as EventListenerOptions;
+        window.addEventListener('scroll', handleScroll, options);
+        window.addEventListener('click', handleClick);
+        return () => {
+            window.removeEventListener('scroll', handleScroll, options);
+            window.addEventListener('click', handleClick);
+        };
+    }, [scrollY]);
```

Comment:
This `useEffect` could have some pretty serious performance issues. Especially only lower end devices. I'm going to try and outline a quick overview why exactly this is a problem and leave a potential solution

1. on initial render, our component mounts and our useEffect gets fired once
2. in that use effect new functions get created and attached as listeners
3. When a user scrolls our event listener is fired and we update our scrollY state.
4. This state update causes the component to rerender
5. The rerender first triggers the cleanup function of our useEffect which removes listeners
6. Then our useEffect is fired, functions created, and listeners get added again.
7. When a user scrolls again repeat steps 3-7

Here's a version of this that doesn't have these performance issues. The main difference here is that `scrollY` is no longer stored in state which means updating it doesn't trigger a rerender of the component. This lack of state also means that our `useEffect` only ever runs on mount therefore not causing a constant stream of adding and removing events

```typescript
const [show, setShow] = useState(true);
const previousScrollYRef = useRef(0);

useEffect(() => {
    const handleScroll = () => {
        setShow(!(window.scrollY > previousScrollYRef.current))
        previousScrollYRef.current = window.scrollY;
    };

    const handleClick = (event: any) => {
        if (!event.target.matches('#navbar, #navbar *')) {
            setShow(false);
        }
    };

    window.addEventListener('scroll', handleScroll);
    window.addEventListener('click', handleClick);

    return () => {
        window.removeEventListener('scroll', handleScroll);
        window.addEventListener('click', handleClick);
    };
}, []);
```

Reply:
Nice code, implemented!